internal/helm/controller/reconcile.go

```diff
@@ -388,9 +389,33 @@ func (r HelmOperatorReconciler) Reconcile(ctx context.Context, request reconcile
 		Manifest: expectedRelease.Manifest,
 	}
 	err = r.updateResourceStatus(ctx, o, status)
+
+	// Parse the reconcile period from CR's annotations
+	annotationsReconcilePeriod := parseReconcilePeriod(o)
+
+	if annotationsReconcilePeriod != "" {
```

Comment:
can we a have a small test that verifies the overriding hierarchy, just to make sure that when a flag is passed the annotation values (if present) override them?